### ISSUE TITLE
Can't start an external process without a timeout interval

### ISSUE THREAD

**Issue by preston-m-price** [fleipold/jproc#4]
Setting a timeout interval of 0ms fails immediately rather that running for as long as necessary.

**Comment by fleipold** [comment]
Are you sure you don't want to set a  (judiciously chosen) timeout for the process? 

In a lot of cases this makes sense. I have been bitten before, e.g. by batch-mode programs that switched back to interactive mode when they hit certain problems ;-).

Currently you could set the timeout  to a very large number to achieve quasi no timeout. I admit that this would however obscure the intent a bit.

Speaking of obscuring intent I prefer not to have magic numbers like 0 or -1 with special semantics.

I would rather add a method to the builder, so that you could write something like this:

~~~ {.java}
ProcResult result = new ProcBuilder("sleep")
        .withArg("0.5")
        .withNoTimeout()
        .run();
~~~

What do you think?

**Comment by fleipold** [comment]
Fixed in 2.2.0.
